package.json

```diff
@@ -21,17 +21,17 @@
     "create-react-component": "yo react-domain"
   },
   "dependencies": {
-    "@theforeman/vendor": "^4.14.0",
+    "@theforeman/vendor": "^6.0.0",
```

Comment:
Would it make sense to pin this to the exact version we have in packaging to ensure dev and production are on the same version?

Reply:
Maybe it would make sense to switch to `~` so it would update to the latest z version but not update the y version until we manually updated.

so `~6.0.0` would pull in 6.0.1, 6.0.2, etc... but not 6.1.0

@sharvit  thoughts?

Also please double check my understanding of tilde and caret ranges here https://docs.npmjs.com/cli/v6/using-npm/semver :)

Reply:
I remember a long discussion where we decided to keep the `^` so we can catch errors in the development environment faster.

Reply:
Any chance you can find that discussion @sharvit ? i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version, so we won't know something is broken until the packages build in the best case or until someone notices something doesn't work in the worst case. 
For example - https://github.com/theforeman/foreman/pull/7505#pullrequestreview-523158457 - tests were green but merging would cause nightly to break and there was no indication that a version bump was needed.

Reply:
Not sure if it's the exact convo, but I know I brought it up here: https://community.theforeman.org/t/pinning-npm-packages-to-y-version/12393

Note that this is before we had the foreman-js meta-package separation, so it could make more sense now to use strict pinning of versions for `@theforeman/*` packages. That way it forces us to update foreman and foreman-packaging to use updated packages in y and z version releases of the foreman-js packages.

Reply:
> Any chance you can find that discussion @sharvit ?

I tried but couldn't find any.

>  i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version

>  i'm worried that developers might forget that we have a different version in production, and tests would be passing for them as well since they run with the latest matching version, so we won't know something is broken until the packages build in the best case or until someone notices something doesn't work in the worst case.

Does it make a difference because it's `@theforeman/*` package or you would suggest that to everything in the `package.json`?

> For example - #7505 (review) - tests were green but merging would cause nightly to break and there was no indication that a version bump was needed.

I am not sure it's a good example because we merge the `helmet` upgrade after we made sure no one using that feature yet and we discussed with the PR author before and with other UI maintainers before merging a releasing as a none breaking changes.
If it was not the case, we would release it as a major release.